lib/iris/tests/stock/_stock_2d_latlons.py

```diff
@@ -0,0 +1,340 @@
+         [-34.99  -46.352 -59.721 -60.34  -47.305 -35.499]
+         [  1.976 -10.626 -22.859 -23.349 -11.595   1.37 ]
+         [ 38.914  25.531  14.312  13.893  24.585  38.215]
+         [ 74.197  60.258  51.325  51.016  59.446  73.268]]
+        >>>
+
+    """
+    def sample_cube(xargs, yargs):
+        # Arguments are to make coord points with np.linspace : start/stop/N.
+        x0, x1, nx = xargs
+        y0, y1, ny = yargs
+        data = np.zeros((ny, nx))
+        data.flat[:] = np.arange(ny * nx) % (nx + 2)  # seems to work ok
+        cube = Cube(data, long_name='test_data')
+        x_pts = np.linspace(x0, x1, nx, endpoint=True)
+        y_pts = np.linspace(y0, y1, ny, endpoint=True)
+        co_x = DimCoord(x_pts, standard_name='longitude', units='degrees')
+        co_y = DimCoord(y_pts, standard_name='latitude', units='degrees')
+        cube.add_dim_coord(co_y, 0)
+        cube.add_dim_coord(co_x, 1)
+        return cube
+
+    if regional:
+        # Extract small region.
+        cube = sample_cube(xargs=(150., 243.75, 6), yargs=(-10., 40., 5))
+
+        # Make contiguous bounds.
+        for ax in ('x', 'y'):
+            cube.coord(axis=ax).guess_bounds()
+    else:
+        # Global data, but drastically reduced resolution.
+
+        cube = sample_cube(xargs=(37.5, 318.75, 6), yargs=(-85., 65., 5))
+
+        # Patch bounds to ensure it is still contiguous + global.
+        for name in ('longitude', 'latitude'):
+            coord = cube.coord(name)
+            coord.guess_bounds()
+            bds = coord.bounds.copy()
+            # Make bounds global, by fixing lowest and uppermost values.
+            if name == 'longitude':
+                bds[0, 0] = 0.0
+                bds[-1, 1] = 360.0
+            else:
+                bds[0, 0] = -90.0
+                bds[-1, 1] = 90.0
+            coord.bounds = bds
+
+    # Get 1d coordinate points + bounds + calculate 2d equivalents.
+    co_1d_x, co_1d_y = [cube.coord(axis=ax).copy() for ax in ('x', 'y')]
+    co_2d_x, co_2d_y = grid_coords_2d_from_1d(co_1d_x, co_1d_y)
+
+    # Remove the old grid coords.
+    for coord in (co_1d_x, co_1d_y):
+        cube.remove_coord(coord)
+
+    # Add the new grid coords.
+    for coord in (co_2d_x, co_2d_y):
+        cube.add_aux_coord(coord, (0, 1))
+
+    if transformed or rotated:
+        # Take the lats + lons as being in a rotated coord system.
+        pole_lat, pole_lon = 75.0, 120.0
+        if transformed:
+            # Reproject coordinate values from rotated to true lat-lons.
+            co_x, co_y = [cube.coord(axis=ax) for ax in ('x', 'y')]
+            # Unrotate points.
+            lons, lats = co_x.points, co_y.points
+            lons, lats = unrotate_pole(lons, lats, pole_lon, pole_lat)
+            co_x.points, co_y.points = lons, lats
+            # Unrotate bounds.
+            lons, lats = co_x.bounds, co_y.bounds
+            shape = lons.shape  # unrotate pole is restricted to 1- or 2-d ??
```

Comment:
If it doesn't support 2d it will be a problem on line 278 where the coordinates will already be 2d

Reply:
It sounds like unrotate_pole can handle 2d arrays. unrotate_pole calls transform_points (in Cartopy) which accepts 1d or 2d (https://github.com/SciTools/cartopy/blob/25c786ebe60f3e7c45c1cfb283a6b150c7c8a931/lib/cartopy/_crs.pyx#L326)

Reply:
Could you remove this comment?

Reply:
It's just that `unrotate_pole` makes an unnecessary check for 1- or 2-d array inputs, whereas it could just work elementwise + not worry about dimensions, which is what we want here.
So I'm flattening + then re-applying the shape afterwards, to support any shape of arrays.

Will try to clarify comments !